packages/rooks/src/hooks/useIntersectionObserverRef.ts

```diff
@@ -26,10 +26,24 @@ function useIntersectionObserverRef(
 
   const [node, setNode] = useState<HTMLElementOrNull>(null);
 
+  const callbackRef = useRef(callback);
+  callbackRef.current = callback;
+
+  const handleIntersectionObserver = useCallback<IntersectionObserverCallback>(
+    (...args) => {
+      return callbackRef.current?.(...args);
+    },
+    []
+  );
+
   useEffect(() => {
     // Create an observer instance linked to the callback function
-    if (node && callback) {
-      const observer = new IntersectionObserver(callback, options);
+    if (node && callbackRef.current) {
```

Comment:
I think we shouldn't be checking for callbackRef.current here. 
```suggestion
    if (node) {
```

Reply:
The `handleIntersectionObserver` will take care of calling the function if it exists :)

Reply:
Thanks! I changed it